### DISCUSSION TITLE
Python 2依赖的workflow在Monterey下如何正常运行

### DISCUSSION THREAD

**Discussion by ZiLoong-L** [alanhg/alfred-workflows#101] [Q&A 常见问题]
非常感谢博主分享的关于Alfred的解答，帮助我解决了很多问题！不知道能不能在博主的Github里提问，但是有一个关于Alfred WorkFlow的问题一直困扰了我很久所以想来请教一下博主！因为新版本MacOS里好像没有了/usr/bin/python这个文件而是用/usr/bin/python3取而代之，这导致许多依赖/usr/bin/python的Alfred WorkFlow插件无法使用了，如下图：
<img width="230" alt="image" src="https://user-images.githubusercontent.com/67189132/163723602-a0ed42fc-122b-4ee4-9248-ac5e1a27eadd.png">
但由于SIP等原因的限制，又无法软链接或重新添加/usr/bin/python文件，不知道博主有没有遇到类似的问题？

**Comment by alanhe421** [comment]
> Python不是我了解的语言，且我基本不依赖py开发的workflow。以下为我想到/查到的方案。仅帮到此。

## 方法1
升级这些workflow，作者可能已经更新支持了新的Python
## 方法2
改造workflow，比如将程序文件单独存储，在执行脚本下拉选择zsh或者bash，内容中明确指定执行环境，比如` /usr/bin/python3 main.py`
## 方法3
安装Python2,且软链为默认环境

```shell
export PATH="/opt/homebrew/bin:/usr/local/bin:${PATH}"
eval "$(brew shellenv)"
brew install pyenv
pyenv install 2.7.18
ln -s "${HOME}/.pyenv/versions/2.7.18/bin/python2.7" "${HOMEBREW_PREFIX}/bin/python"
```

**Comment by ZiLoong-L** [comment]
非常感谢博主的解答！
